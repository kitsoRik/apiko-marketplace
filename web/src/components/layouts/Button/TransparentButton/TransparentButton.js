import React from 'react';

import "./TransparentButton.scss";

const TransparentButton = ({ className, value, disabled = false, darkMode = null, ...props }) => {
  return (
    <button
      className={`button-transparent ${className ?? ""}`}
      disabled={disabled ? "true" : null}
      dark-mode={darkMode ? "true" : null}
      {...props}>
      {value}
    </button>
  )
};

export default TransparentButton;