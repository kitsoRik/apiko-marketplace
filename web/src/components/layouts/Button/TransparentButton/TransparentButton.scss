.button-transparent {
	color: #000;
	background-color: transparent;

	border: none;
	border-radius: 4px;
	text-decoration: none;

	transition: 0.1s linear;
	transition-property: color background-color;

	&[dark-mode="true"] {
		color: #fff;

		&:hover {
			background-color: #fff6;
		}
	}

	&:hover {
		background-color: #0002;
	}

	&:disabled {
		background-color: gray;
	}

	&:active {
		background-color: #0001;
	}
}
