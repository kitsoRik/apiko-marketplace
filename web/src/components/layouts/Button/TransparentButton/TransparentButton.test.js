import React from 'react';
import { create } from 'react-test-renderer';
import TransparentButton from './TransparentButton';


describe("TransparentButton", () => {
    it("should render without value", () => {
        const component = create(<TransparentButton />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should render with value", () => {
        const component = create(<TransparentButton value="My custom value" />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should render with className", () => {
        const component = create(<TransparentButton className="my-custom-class-name" />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should render with darkMode=true", () => {
        const component = create(<TransparentButton value="My custom value" darkMode={true} />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should render with darkMode=false", () => {
        const component = create(<TransparentButton value="My custom value" darkMode={false} />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should render disabled", () => {
        const component = create(<TransparentButton value="My custom value" disabled={true} />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
});