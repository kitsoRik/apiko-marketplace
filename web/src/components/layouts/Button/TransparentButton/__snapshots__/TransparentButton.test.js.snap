// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransparentButton should render disabled 1`] = `
<button
  className="button-transparent "
  dark-mode={null}
  disabled="true"
>
  My custom value
</button>
`;

exports[`TransparentButton should render with className 1`] = `
<button
  className="button-transparent my-custom-class-name"
  dark-mode={null}
  disabled={null}
/>
`;

exports[`TransparentButton should render with darkMode=false 1`] = `
<button
  className="button-transparent "
  dark-mode={null}
  disabled={null}
>
  My custom value
</button>
`;

exports[`TransparentButton should render with darkMode=true 1`] = `
<button
  className="button-transparent "
  dark-mode="true"
  disabled={null}
>
  My custom value
</button>
`;

exports[`TransparentButton should render with value 1`] = `
<button
  className="button-transparent "
  dark-mode={null}
  disabled={null}
>
  My custom value
</button>
`;

exports[`TransparentButton should render without value 1`] = `
<button
  className="button-transparent "
  dark-mode={null}
  disabled={null}
/>
`;
